Doc/c-api/unicode.rst

```diff
@@ -1399,13 +1399,34 @@ They all return ``NULL`` or ``-1`` if an exception occurs.
    set.  Separators are not included in the resulting list.
 
 
+.. c:function:: PyObject* PyUnicode_RSplit(PyObject *unicode, PyObject *sep, Py_ssize_t maxsplit)
+
+   Similar to :c:func:`PyUnicode_Split`, but splitting will be done beginning
+   at the end of the string.
+
+
 .. c:function:: PyObject* PyUnicode_Splitlines(PyObject *unicode, int keepends)
 
    Split a Unicode string at line breaks, returning a list of Unicode strings.
    CRLF is considered to be one line break.  If *keepends* is ``0``, the Line break
    characters are not included in the resulting strings.
 
 
+.. c:function:: PyObject* PyUnicode_Partition(PyObject *unicode, PyObject *sep)
+
+   Split the string at the first occurrence of *sep*, and return a 3-tuple
+   containing the part before the separator, the separator itself, and the part
+   after the separator. If the separator is not found, return a 3-tuple
+   containing the string itself, followed by two empty strings.
```

Comment:
Maybe we can say that it's equivalent to `str.partition()` (idem for the othre functions, we could link them to their Python equivalent).

Reply:
Not sure. The other c-functions don't link to their python equivalents either (`PyUnicode_Split`, `PyUnicode_Splitlines`, `PyUnicode_Join`).

To be fair though, I mostly reused the Python docstring here.
https://docs.python.org/3.13/library/stdtypes.html#str.partition

Reply:
In general, we say "Unicode string" in the C documentation but don't say it in the Python one as `str` is always unicode. The reason why I suggested mentioning the Python method was in hope that there were examples, but it appears not so no need to mention them.